xbmc/android/activity/XBMCApp.cpp

```diff
@@ -101,6 +102,7 @@ CXBMCApp::CXBMCApp(ANativeActivity* nativeActivity)
   : CJNIApplicationMainActivity(nativeActivity)
   , CJNIBroadcastReceiver("org/xbmc/kodi/XBMCBroadcastReceiver")
 {
+  m_xbmcappinstance = this;
```

Comment:
Well, we "hide" our standard singleton in a clever macro/template, but I honestly don't see how this is conceptually uglier. At the end of the day, it's still a static variable, and at least this implem doesn't make it unnecessarily global.

This whole "50% stack singleton/50% static" class is ugly really ;)